lib/build.gradle.kts

```
@@ -55,6 +69,104 @@ tasks.dokkaJavadoc {
+                """(?:public|private|protected)?\s+[\w<>\[\]]+\s+$functionName\s*\([^)]*\)\s*\{([\s\S]*?)}\s*""",
+                RegexOption.MULTILINE,
+            )
+        }
+
+    }
+}
+
+tasks.register("generateGuides") {
+    doLast {
+        val kotlinSamplesDir = "${project.projectDir}/src/test/kotlin"
+        val javaSamplesDir = "${project.rootDir}/javademo/src/main/java/com/example/javademo"
+        val guideFile = file("guide.md")
+        val kotlinGuideFile = project.projectDir.resolve("${Language.KOTLIN.langName}-guide.md")
+        val javaGuideFile = project.projectDir.resolve("${Language.JAVA.langName}-guide.md")
+
+        val guideContent = guideFile.readText()
+
+        val docsStartRegex = Regex("""^\s*//\s*docsStart\s*$""")
+        val docsEndRegex = Regex("""^\s*//\s*docsEnd\s*$""")
+
+        fun extractCodeSection(path: String, functionName: String, language: Language): String {
+            val file = file(path)
+            if (!file.exists()) {
+                return ""
+            }
+
+            val content = file.readText()
+
+            val matchResult = language.getFunctionRegex(functionName).find(content) ?: return ""
+            val codeSection = matchResult.groupValues[1].lines()
+            val filteredLines = mutableListOf<String>()
+            var capture = false
+
+            for (line in codeSection) {
+                when {
+                    docsStartRegex.containsMatchIn(line) -> capture = true
+                    docsEndRegex.containsMatchIn(line) -> capture = false
+                    capture -> filteredLines.add(line)
+                }
+            }
+            val minIndent = codeSection.filter { it.isNotBlank() }
+                .minOfOrNull { it.indexOfFirst { char -> !char.isWhitespace() } } ?: 0
+
+            return filteredLines.joinToString("\n") { if (it.isBlank()) it else it.drop(minIndent) }
+        }
+
+        fun processFileContent(dir: String, language: Language): String {
+            var content = language.getCodeSectionRegex().replace(guideContent) { matchResult ->
+                val path = matchResult.groupValues[1]
+                val functionName = matchResult.groupValues[2]
+                val codeSection = extractCodeSection(path="$dir/$path", functionName=functionName, language=language)
+                if (codeSection.isBlank()) "" else "```$language\n$codeSection\n```"
+            }
+
+            content = content.replace(Language.KOTLIN.getCodeSectionRegex(), "")
+                .replace(Language.JAVA.getCodeSectionRegex(), "")
+
+            return content.replace(
+                if (language == Language.KOTLIN) Language.JAVA.getCodeBlockRegex() else Language.KOTLIN.getCodeBlockRegex(),
+                ""
+            )
+        }
+
+        val kotlinContent = processFileContent(dir=kotlinSamplesDir, language = Language.KOTLIN)
+        val javaContent = processFileContent(dir=javaSamplesDir, language = Language.JAVA)
```

```suggestion
        val kotlinContent = processFileContent(dir = kotlinSamplesDir, language = Language.KOTLIN)
        val javaContent = processFileContent(dir = javaSamplesDir, language = Language.JAVA)
```

Even better, format the entire file, as this is not checked by the checks.

---

Also not sure named arguments here and above are really necessary since argument names are self-explanatory by themselves, but this is really far on nit side, so it's up to you 😄